### ISSUE TITLE
Enable opt-level=2 for failing vLLM models

### ISSUE THREAD

**Issue by vkovacevicTT** [tenstorrent/tt-xla#5543]
Some vLLM models were lowered to opt-level=0 due to DRAM OOM on n150 machines:
Llama-3.1-8B, Ministral-8B and Qwen3-8B.

We should fix the error and enable opt-level=2.

**Comment by rpavlovicTT** [comment]
## Root cause (reproduced on n150, `llama-3.1-8b`)

OOM at warm-up (opt-2, `cpu_sampling=True`):
```
TT_FATAL: Out of Memory: Not enough space to allocate 1050673152 B DRAM buffer across 12 banks
(allocated: 999238944 B/bank, free: 71534240 B/bank) — bank_manager.cpp:462
```

Failing op = the **lm_head bf16→bfp_bf8 const-eval conversion**. lm_head (`128256×4096`, excluded from bfp8) is tilized **on device**: `to_device` → DRAM bf16 row-major (1.05 GB) → **`to_layout` → DRAM bf16 tiled (+1.05 GB) ← OOM** → `from_device` → `typecast`→bfp8 (host) → `to_device` → DRAM bfp8 (~0.56 GB). Both bf16 copies are live at the tilize → **2.1 GB transient** on ~10 GB already resident. `1050673152 = 128256 × 4096 × 2`.

## Isolated: it's the `cpu_sampling=True` graph structure, not opt-2

| opt | cpu_sampling | path | result |
|--|--|--|--|
| 0 | False | fused | PASS |
| 1 | False | fused | PASS |
| 2 | **True** | unfused | **OOM** |
| 2 | **False** | fused | **PASS** (486 s, device-sampled, 12.2 samples/s) |

**opt-2 itself is fine.** `cpu_sampling=True` emits a *separate* compute-logits graph whose lm_head const-eval fires when the full weight + KV working set is already resident, so the 2.1 GB spike overflows. The fused path (`cpu_sampling=False`) keeps lm_head in the decode graph and the conversion happens with headroom.

Correction to my earlier note: `cpu_sampling=False` at opt≥1 is **not** blocked by #4387 — both opt-1 and opt-2 fused runs compiled and generated output (gather_logprobs just falls back to CPU at runtime).

## Unblock

Run these models at **opt-2 with `cpu_sampling=False`** (on-device sampling) — verified working for `llama-3.1-8b`. Caveat: the device sampler doesn't support per-row seeds (`platform.py`), which is fine for greedy benchmarking but a limitation for seeded sampling. The underlying const-eval 2× bf16 transient is still latent (it re-materializes both bf16 copies in DRAM on device; host-side tilize isn't possible) and would resurface on larger models or via the unfused path — no compiler fix for that yet.

## Concrete change + open question

Change: drop `optimization_level=0` and the forced `cpu_sampling=True` for `llama-3.1-8b` / `ministral-8b` / `qwen3-8b` in the single-device benchmark set and let them run opt-2 + `cpu_sampling=False`.

Before doing that, resolve **why `cpu_sampling=True` is forced at opt≥1** (`tests/benchmark/test_vllm_benchmarks.py:69-70`). The referenced "TTConfig raises if enable_trace && opt≥1 && !cpu_sampling" constraint is **not** enforced in current `platform.py`, and device sampling now compiles and runs at opt-1 and opt-2. So the forcing looks stale/over-conservative — unless it's intentionally retained for seeded-sampling correctness or perf parity. Confirming that is the remaining gate before flipping the configs.

**Comment by rpavlovicTT** [comment]
hey @kmabeeTT @vkovacevicTT , has anyone had chance to pick this up?

**Comment by vkovacevicTT** [comment]
@rpavlovicTT 
Sampling on device is on main: https://github.com/tenstorrent/tt-xla/commit/7963edc84bc90b00d13c9653a1203842776e4e30
Ministral-8B and Qwen3-8B pass nightly (fail on check regression job).
Verified Llama-3.1-8B locally, we can enable opt-level=2 for it.
